seqr/views/apis/anvil_workspace_api_tests.py

```diff
@@ -552,6 +598,118 @@ def test_create_project_from_workspace(self, mock_tempfile, mock_mv_file, mock_s
         self.assertEqual(response.status_code, 302)
         self.assertEqual(response.url, '/login/google-oauth2?next=/api/create_project_from_workspace/my-seqr-billing/anvil-no-project-workspace2/submit')
 
+    @mock.patch('seqr.models.Project._compute_guid', lambda project: f'P_{project.name}')
+    @mock.patch('seqr.views.apis.anvil_workspace_api.id_token.fetch_id_token', lambda *args: MOCK_TOKEN)
+    @mock.patch('seqr.views.utils.airtable_utils.AIRTABLE_API_KEY', MOCK_AIRTABLE_KEY)
+    @mock.patch('seqr.views.utils.airtable_utils.AIRTABLE_URL', MOCK_AIRTABLE_URL)
+    @mock.patch('seqr.views.apis.anvil_workspace_api.AIRFLOW_WEBSERVER_URL', MOCK_AIRFLOW_URL)
+    @mock.patch('seqr.views.apis.anvil_workspace_api.BASE_URL', 'http://testserver/')
+    @mock.patch('seqr.views.apis.anvil_workspace_api.ANVIL_LOADING_DELAY_EMAIL', None)
+    @mock.patch('seqr.views.apis.anvil_workspace_api.datetime')
+    @mock.patch('seqr.views.apis.anvil_workspace_api.logger')
+    @mock.patch('seqr.views.apis.anvil_workspace_api.load_uploaded_file')
+    @mock.patch('seqr.views.apis.anvil_workspace_api.send_html_email')
+    @mock.patch('seqr.views.apis.anvil_workspace_api.safe_post_to_slack')
+    @mock.patch('seqr.views.apis.anvil_workspace_api.mv_file_to_gs')
+    @mock.patch('seqr.views.apis.anvil_workspace_api.tempfile.NamedTemporaryFile')
+    @responses.activate
+    def test_add_workspace_data(self, mock_tempfile, mock_mv_file, mock_slack, mock_send_email, mock_load_file,
```

Comment:
a ton of this function is copy  pasted from the create test. It would be better to make more helper functions and share more code. I think a good idea might be to actually put these 2 functions in a different test class than the other test, something like `class LoadAnvilDataAPITest(AnvilAuthenticationTestCase):` . Then you could add all the shared mocks to the class instead of both functions, you could write a custom `setUp` function that does the responses set up, and the scope for the helper functions would be neater